File: wwwroot/src/interfaces/pokemonEval.ts
export interface PokemonEval{
    health: number;
    defense: number;
    weight: number;
    attack: number;
    speed: number;
    height: number;
    special_attack: number;
    special_defense: number;
    id: number;
    name: string;
    true_value: string;
    knn_prediction: string;
    log_reg_prediction: string;
    tree_prediction: string;
}

export interface Metrics{
    f1_score: number;
    accuracy_score: number;
    precision_score: number;
    recall_score: number;
}

export interface PokemonEvalResponse{
    model_eval: PokemonEval[];
    knn_metrics: Metrics;
    log_reg_metrics: Metrics;
    tree_metrics: Metrics;
}

File: Models/PokemonEval.cs
using System.Text.Json.Serialization;

namespace PokemonMlEvalWebApp.Models;

public class PokemonEval
{
    [JsonPropertyName("health")]
    public int Health {get; set;}
    [JsonPropertyName("defense")]
    public int Defense {get; set;}
    [JsonPropertyName("weight")]
    public int Weight {get; set;}
    [JsonPropertyName("attack")]
    public int Attack {get; set;}
    [JsonPropertyName("speed")]
    public int Speed {get; set;}
    [JsonPropertyName("height")]
    public int Height {get; set;}
    [JsonPropertyName("special_attack")]
    public int SpecialAttack {get; set;}
    [JsonPropertyName("special_defense")]
    public int SpecialDefense {get; set;}
    [JsonPropertyName("id")]
    public int Id {get; set;}
    [JsonPropertyName("name")]
    public string Name {get; set;} = string.Empty;
    [JsonPropertyName("true_value")]
    public string TrueType {get; set;} = string.Empty;
    [JsonPropertyName("knn_prediction")]
    public string KNN {get; set;} = string.Empty;
    [JsonPropertyName("log_reg_prediction")]
    public string LogReg {get; set;} = string.Empty;
    [JsonPropertyName("tree_prediction")]
    public string Tree {get; set;} = string.Empty;
}